pkg/controller/controller.go

```diff
@@ -833,3 +838,28 @@ func (c *Controller) DeleteVolumeGroup(contx context.Context, opt *pb.DeleteVolu
 
 	return pb.GenericResponseResult(nil), nil
 }
+func (c *Controller) GetMetrics(context context2.Context, opt *pb.GetMetricsOpts) (*pb.GenericResponse, error) {
+	log.Info("in controller get metrics methods")
+
+	var result *[]model.MetricSpec
+	var err error
+
+	if opt.StartTime=="" && opt.EndTime==""{
+		// no start and end time specified, get the latest value of this metric
+		result, err = c.metricsController.GetLatestMetrics(opt)
+	} else if opt.StartTime==opt.EndTime{
+		// same start and end time specified, get the value of this metric at that timestamp
+		result, err = c.metricsController.GetInstantMetrics(opt)
+	} else {
+		// range of start and end time is specified
+		result, err = c.metricsController.GetRangeMetrics(opt)
+	}
+
+	if err != nil {
+		log.Error("GetMetrics failed: ", err.Error())
```

Comment:
All error log message should begin with lower letter.

Reply:
updated.